### ISSUE TITLE
How can i build a Multi Label classification CatBoost Model?

### ISSUE THREAD

**Issue by OmarVivas** [catboost/catboost#1420]
Problem: How can i build a Multi Label classification CatBoost Model in R?
catboost version: 0.24.1
Operating System: Windows
CPU: Yes

I currently parcipate in a Kaggle competition (https://www.kaggle.com/c/lish-moa) and I need to build 206 model (one per each feature)... I can build a Multi Label classification CatBoost Model in R?... Is there an example?.

I'm a CatBoost fan, it's a great model.
 
Thanks

**Comment by ThomasWolf0701** [comment]
I am also trying to figure this out. The predict function of the R package mentions multilabel classification, but the train function does not and there also no multilabel losses. But it is possible to pass a label matrix.

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
In theory, it should suffices to pass loss_function='MultiClass' to train.

https://github.com/catboost/catboost/blob/8305d3c0a383c981e6cf08fc32ab9c52acb7fe11/catboost/R-package/R/catboost.R#L1429

**Comment by ThomasWolf0701** [comment]
> 
> 
> In theory, it should suffices to pass loss_function='MultiClass' to train.
> 
> https://github.com/catboost/catboost/blob/8305d3c0a383c981e6cf08fc32ab9c52acb7fe11/catboost/R-package/R/catboost.R#L1429


But this would only apply to multiple classes, not multilabel/multiregression where each sample has multiple values/labels )

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
For multiregression models, loss_function should be 'MultiRMSE' and the dataset should be specified by a file + column-description file with multiple Label columns.

**Comment by OmarVivas** [comment]
Hi Evgueni, is it work to R ( MultiRMSE)?... Could you send us an
example?... I think, CatBoost is great and get better every month.
Thanks you.
Omar Vivas

**Comment by ThomasWolf0701** [comment]
> 
> 
> For multiregression models, loss_function should be 'MultiRMSE' and the dataset should be specified by a file + column-description file with multiple Label columns.

When I do  help(catboost.train)  MultiRMSE is not among the supported loss functions (Documentation for package ‘catboost’ version 0.24.2)

**Comment by Sphinxs** [comment]
There's nothing for multi-label classification, like [binary cross-entropy with logits](https://gombru.github.io/2018/05/23/cross_entropy_loss/)? (coming soon? cause I read the docs and there's nothing)

**Comment by sridharnomula** [comment]
We are really looking for multilabel classification model from catboost with python API

**Comment by kizill** [comment]
Released in 1.0.0

**Comment by adelchi91** [comment]
@kizill indeed, it was released, but I cannot seem to find example and or tutorial. The documentation is also hard to find.

**Comment by cccntu** [comment]
There is a tutorial here: https://github.com/catboost/tutorials/blob/94e3bc9c3510ed69c2da7f544ec8979387419ec9/classification/multilabel_classification_tutorial.ipynb

**Comment by RucGuipeng** [comment]
> There is a tutorial here: https://github.com/catboost/tutorials/blob/94e3bc9c3510ed69c2da7f544ec8979387419ec9/classification/multilabel_classification_tutorial.ipynb

Thanks for your tutorial, now I want to use a custom defined loss, but it seems that I can not used custom loss with multilabel(Y is a matrix), how can I train a Catboost model under this situation?

**Comment by RucGuipeng** [comment]
![image](https://user-images.githubusercontent.com/50044043/191449058-6940f5b4-1a63-420f-9e74-be40376cbe15.png)
![image](https://user-images.githubusercontent.com/50044043/191449100-4d4e683d-b2ea-4bf7-9e24-441a27e3b5a3.png)

**Comment by ek-ak** [comment]
Hi! You need  MultiTargetCustomObjective, you can example  in tutorial https://github.com/catboost/tutorials/blob/master/custom_loss/custom_loss_and_metric_tutorial.ipynb
